ocaml/typing/types.ml

```diff
@@ -738,7 +751,7 @@ let log_change ch =
 
 let () =
   Mode.set_append_changes (fun changes -> log_change (Cmodes !changes));
-  Jkind.Sort.change_log := (fun change -> log_change (Csort change))
+  !jkind_sort_change_log := (fun change -> log_change (Csort change))
```

Comment:
This is pretty convoluted -- and I think it's broken.

Convoluted: we have a ref to a ref, with one set in `Jkind.Sort` and then the other one set here. Basically, I think the old setup was because Types depended on Jkind, and now that's been flipped -- but instead of undoing the old treatment, now there's another step.

Broken: I think that `jkind_sort_change_log` won't be set by `Jkind.Sort` before this assignment here happens, meaning that the assignment to `jkind_sort_change_log` in `Jkind.Sort` effectively overwrites this change, and it makes `Jkind.Sort.log_change` into a no-op. I'm a little surprised (but not very) that this doesn't show up in tests.

The fix, I think, is to define the (very simple) sort-related change-log functions here in types, breaking the recursion. Maybe there's another way, but I don't see an easy one that doesn't export the `log_change` function immediately above.